integration-tests/gradle/src/test/java/io/quarkus/gradle/FastJarFormatWorksTest.java

```diff
@@ -50,20 +49,22 @@ public void testFastJarFormatWorks() throws Exception {
             assertThatOutputWorksCorrectly(logs);
 
             // test that the application name and version are properly set
-            assertThat(DevModeTestUtils.getHttpResponse("/hello", getQuarkusDevBrokenReason()).equals("hello"));
+            assertThat(DevModeTestUtils.getHttpResponse("/hello", () -> {
+                return jarRun == null ? null : jarRun.isDone() ? "jar run mode has terminated" : null;
+            }).equals("hello"));
         } finally {
             process.destroy();
         }
     }
 
-    static void assertThatOutputWorksCorrectly(String logs) {
+    private void assertThatOutputWorksCorrectly(String logs) {
         assertThat(logs.isEmpty()).isFalse();
         String infoLogLevel = "INFO";
         assertThat(logs.contains(infoLogLevel)).isTrue();
         assertThat(logs.contains("cdi, resteasy")).isTrue();
     }
 
-    private void dumpFileContentOnFailure(final Callable<Void> operation, final File logFile,
+    protected static void dumpFileContentOnFailure(final Callable<Void> operation, final File logFile,
```

Comment:
This isn't a good idea since you introduce unnecessary coupling between the test classes. It would be better if you moved this method to `LaunchUtils`

Reply:
Okk